overlord/snapstate/snapstate.go

```diff
@@ -245,6 +245,63 @@ func Update(s *state.State, name, channel string, userID int, flags Flags) (*sta
 	return doInstall(s, &snapst, ss)
 }
 
+// Enable sets a snap to the active state
+func Enable(s *state.State, name string) (*state.TaskSet, error) {
+	var snapst SnapState
+	err := Get(s, name, &snapst)
+	if err != nil && err != state.ErrNoState {
```

Comment:
If it's NoState, the snap doesn't exist. We probably shouldn't try to enable that and instead return "snap %q not found" or whatever we do in similar cases.